docs/docs-beta/docs/guides/deployment/kubernetes.md

```diff
@@ -1,4 +1,221 @@
+
+</details>
+
+
+## Step 0a: Understand the Dagster deployment architecture
+Do we need this part? Should it be a pre-req that we link to?
+
+## Step 0b: Example project tour
+If you are deploying your own Dagster project, skip ahead to Step 1. However, if you are using the example project for this guide, this step will walk you through the contents of the project.
+
+The example project should contain the following files
+```bash
+TODO FILE TREE
+```
+
+The Dagster project can be found in `iris_analysis`. The project itself is found in the `iris_analysis/__init__.py` file. This file contains a single asset that downloads a dataset about iris flowers and logs the number of rows.
+
+The example project also contains a `workspace.yaml` file and a `Dockerfile`. These files will be covered in the following steps.
+
+## Step 1: Get your Dagster project ready to deploy
+You will need to add a `workspace.yaml` file to your Dagster project to be ready to deploy with Kubernetes. If you are using the [example project](/todo), you will already have this file.
+
+The `workspace.yaml` file tells Dagster where to find the `Definitions` object in your project. This file should be at the LOCATION of your project.
+
+<CodeExample filePath="guides/deployment/kubernetes/workspace.yaml" language="yaml" title="Example workspace.yaml" />
+For example, in the sample `iris_analysis` project, the `Definitions` object is found in the `iris_analysis/__init__.py` file. The `workspace.yaml` tells Dagster to load this project and give it the name `iris_analysis`.
+
+You can also point to a python module in the `workspace.yaml` file. To learn more about writing a `workspace.yaml` see the [Workspace](/todo) guide.
+
+## Step 2: Write and build a Docker image containing your Dagster project
+### Step 2.1: Write a Dockerfile
+You will need to build a Docker image that contains your Dagster project and all of its dependencies. The Dockerfile should copy your Dagster project and the `workspace.yaml` file created in Step 1 into the image and install `dagster`, `dagster-postgres`, and `dagster-k8s`, along with any other libraries your project depends on. Finally, ensure that port 80 is exposed. This will be used to set up port-forwarding later.
+
+<CodeExample filePath="guides/deployment/kubernetes/Dockerfile" language="docker" title="Example Dockerfile" />
+The example project has a dependency on `pandas` so it is included in the `pip install` command.
+
+### Step 2.2: Build and push a Docker image
+
+To build your Docker image, run the following command from the directory where your Dockerfile is located:
+
+```bash
+docker build . -t iris_analysis:1
+```
+This builds the Docker image from Step 2.1 and gives it the name `iris_analysis` and tag `1`. You can set custom values for both the name and the tag. We recommend that each time you rebuild your Docker image, you assign a new value for the tag to ensure that the correct image is used when running your code.
```

Comment:
for the tech reviewer: i'd like to spot check that this statement is correct

Reply:
paging dr @jmsanders

Reply:
That is correct. I also wonder how much prior knowledge we want to assume here? Or how much we want to offload some of the pre-req stuff to third party docs (specifically the Docker and/or K8s docs)?

I think it's fair to assume if you're going to manage your own open source deployment of Dagster on K8s, you:
- have basic familiar with containers
- have basic familiar with k8s
- or are comfortable reading up on those two things first